### DISCUSSION TITLE
User concerns over disabled menu item contrast

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI#1967] [General]
Some users have reported that they are having difficulty distinguishing between disabled and enabled sub-menu options. Can we review the contrast settings on this text?

It is helpful that the cursor changes to disabled when over a disabled item.

![image](https://github.com/iTwin/iTwinUI/assets/15036346/c9a2db1a-8cab-4fcc-bd7b-ed6e86b458d2)

**Comment by mayank99** [comment]
This is a general problem with disabled controls (and placeholder text too). We have to make a tradeoff between:
- disabled text not having high enough contrast (and therefore being difficult to read, especially for low-vision users)
- disabled text not having enough contrast difference from the regular text (and therefore being difficult to distinguish).

We haven't figured out a solution yet, but it's something we're aware of.

Our current recommendation is that you use words to indicate that something is disabled, and maybe also explain why it is disabled and/or how to make it not disabled.

For example, the sublabel for **"Next"** could say "Not available because ___" or "Not available. Please perform ___ action". If you want to make it more obvious, the label itself could say "Next (not available)"

**Reply by HaveSpacesuit** [reply]
These are good suggestions. I'll see if they make sense for our team. Thanks!

**Reply by mayank99** [reply]
**Update**: We have adjusted the colors of disabled controls in [3.14.0](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.14.0) and introduced a new hover state that to keep them accessible.


https://github.com/user-attachments/assets/e9b22945-031e-44a7-99a7-848337a70aea


My previous suggestions are still applicable and will work well with this change.
